runtime/scripts/parts/patternmatch.js

```diff
@@ -76,7 +78,8 @@ PatternMatchPart.prototype = /** @lends Numbas.PatternMatchPart.prototype */ {
 		displayAnswerString: '',
 		displayAnswer: '',
 		caseSensitive: false,
-		partialCredit: 0
+		partialCredit: 0,
+		matchMode: 'regex'
 	},
 
 	/** Compute the correct answer, based on the given scope
```

Comment:
You will need to change `getCorrectAnswer` to only add the `^` and `$` characters if `this.settings.matchMode == 'regex'`